service/history/replication/task_executor.go

```diff
@@ -147,43 +147,45 @@ func (e *taskExecutorImpl) handleActivityTask(
 	}
 
 	err = syncActivityAction()
+	retryErr, ok := toRetryTaskV2Error(err)
+	if !ok {
+		return err
```

Comment:
in existing implementation `syncActivityAction()` is still called for not `ok` cases at line 189

Reply:
I think that's a bug.

Reply:
Synced offline, we are in an agreement that this is a bug.